mne_bids/write.py

```diff
@@ -1354,3 +1355,150 @@ def write_anat(bids_root, subject, t1w, session=None, acquisition=None,
+
+    if bids_path.root is None:
+        raise ValueError('The root of the "bids_path" must be set. '
+                         'Please use `bids_path.update(root="<root>")` '
+                         'to set the root of the BIDS folder to read.')
+
+    # Read raw and sidecar file.
+    raw = read_raw_bids(bids_path=bids_path,
+                        extra_params=dict(allow_maxshield=True, preload=True),
+                        verbose=False)
+    bads_raw = raw.info['bads']
+
+    channels_fname = _find_matching_sidecar(bids_path, suffix='channels',
+                                            extension='.tsv')
+    tsv_data = _from_tsv(channels_fname)
+    if 'status' in tsv_data:
+        bads_tsv = _get_bads_from_tsv_data(tsv_data)
+    else:
+        bads_tsv = []
+
+    # Read sidecar and create required columns if they do not exist.
+    if 'status' not in tsv_data:
+        logger.info('No "status" column found in input file. Creating.')
+        tsv_data['status'] = ['good'] * len(tsv_data['name'])
+
+    if 'status_description' not in tsv_data:
+        logger.info('No "status_description" column found in input file. '
+                    'Creating.')
+        tsv_data['status_description'] = ['n/a'] * len(tsv_data['name'])
+
+    # Update the sidecar data.
+    if overwrite:
+        # XXX In cases where the "status" and / or "status_description"
+        # XXX columns were just created by us, we overwrite them again
+        # XXX here. This is not optimal in terms of performance, but
+        # XXX probably doesn't hurt anyone.
+        logger.info('Resetting status and description for all channels.')
+        tsv_data['status'] = ['good'] * len(tsv_data['name'])
+        tsv_data['status_description'] = ['n/a'] * len(tsv_data['name'])
+    elif not overwrite and not bads_tsv and bads_raw:
+        # No bads are marked in channels.tsv, but in raw.info['bads'], and
+        # the user requested to UPDATE (overwrite=False) the channel status.
+        # -> Inject bads from info['bads'] into channels.tsv before proceeding!
+        for ch_name in bads_raw:
+            idx = tsv_data['name'].index(ch_name)
+            tsv_data['status'][idx] = 'bad'
+
+    # Now actually mark the user-requested channels as bad.
+    for ch_name, description in zip(ch_names, descriptions):
+        if ch_name not in tsv_data['name']:
+            raise ValueError(f'Channel {ch_name} not found in dataset!')
+
+        idx = tsv_data['name'].index(ch_name)
+        logger.info(f'Processing channel {ch_name}:\n'
+                    f'    status: bad\n'
+                    f'    description: {description}')
+        tsv_data['status'][idx] = 'bad'
+        tsv_data['status_description'][idx] = description
+
+    _write_tsv(channels_fname, tsv_data, overwrite=True, verbose=verbose)
+
+    # Update info['bads']
+    bads = _get_bads_from_tsv_data(tsv_data)
+    raw.info['bads'] = bads
+    # XXX (How) will this handle split files?
```

Comment:
what is your concern?

Reply:
Oh it's very easy, I've never worked with split files before. So here (actually in the next line, not shown here by github) we're only updating the very first file -- `raw.filenames[0]`. In my current tests, there is always exactly ONE file in `raw.filenames`. If we have split files, it could be multiple. I'm not sure if we would have to iterate over all of them, or if only the first one carries a (meaningful) `info` dict, and updating that one is sufficient.

Reply:
for a split file then sidecar files should be identical. It's basically one file split over different files. metadata should be therefore exactly the same.